main.go

```diff
@@ -0,0 +1,239 @@
+	"github.com/companieshouse/kafka-management-tool/schemas"
+)
+
+// Assigns all flags to variables
+var (
+	brokerPtr         = flag.String("broker", "", "Broker address")
+	topicPtr          = flag.String("topic", "", "Topic name")
+	schemaPtr         = flag.String("schema", "", "Schema name")
+	schemaRegistryPtr = flag.String("schema-registry", "", "Schema Registry")
+	partitionPtr      = flag.Int64("partition", 0, "Partition (default: 0)")
+	offsetPtr         = flag.String("offset", "", "Offset number")
+	jsonOutPtr        = flag.Int64("json-out", 0, "Print deserialized JSON message (default: 0)")
+)
+
+func main() {
+	flag.Parse()
+
+	// check if all mandatory flags have been provided
+	validateFlags()
+
+	// Print flags and parameters of the tool
+	printFlags()
+
+	// create offset array
+	offsetArray := createOffsetArray(*offsetPtr)
+
+	// create default config for sarama
+	config := sarama.NewConfig()
+
+	//create consumer
+	consumer, err := sarama.NewConsumer([]string{*brokerPtr}, config)
+	if err != nil {
+		panic(err)
+	}
+
+	// create messages chan
+	messages := make(chan *sarama.ConsumerMessage)
+	go consumePartition(consumer, *topicPtr, int32(*partitionPtr), messages, offsetArray)
+
+	// create signals chan
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	consumed := 0
+ConsumerLoop:
+	for {
+		select {
+		case msg := <-messages:
+
+			// make messageBytes channel
+			messageBytes := make(chan []byte)
+
+			// if jsonOutPtr is set to 1, then output JSON and republish
+			// otherwise just republish message
+			if *jsonOutPtr == 1 {
+				go outputJSON(msg, messageBytes)
+			} else {
+				go func() {
+					messageBytes <- msg.Value
+				}()
+			}
+
+			// create producer message
+			producerMessage := &producer.Message{
+				Value: <-messageBytes,
+				Topic: *topicPtr,
+			}
+
+			//create new producer
+			p, err := producer.New(&producer.Config{Acks: &producer.WaitForAll, BrokerAddrs: []string{*brokerPtr}})
+			if err != nil {
+				fmt.Println("error creating producer")
+				panic(err)
+				os.Exit(1)
+			}
+
+			// republish message
+			fmt.Printf("Republishing message for offset: %v \n", msg.Offset)
+			partition, offset, err := p.Send(producerMessage)
+			if err != nil {
+				fmt.Println("error republishing message with offset: %v, \n", msg.Offset)
+				panic(err)
+				os.Exit(1)
+			}
+			fmt.Printf("Successfully republished message with offset %v to topic: %v using partition: %v new offset: %v \n", msg.Offset, *topicPtr, partition, offset)
+			fmt.Println("---------------------")
+
+			consumed++
+
+			// check if we have hit the end of the array of messages
+			// if so, exit the tool by breaking the ConsumerLoop
+			if consumed == len(offsetArray) {
+				break ConsumerLoop
+			}
+		case <-signals:
+			fmt.Println("break")
+			break ConsumerLoop
+		}
+	}
+}
+
+func outputJSON(msg *sarama.ConsumerMessage, messageBytes chan []byte) {
+	schemaStruct := schemas.IdentifySchema(*topicPtr)
```

Comment:
do we expect every topic we use to have an avro schema assigned to it?

if so then we should be checking for a `nil` value returned here and error.

otherwise the code below should allow for topics without avro schemas and skip the avro schema processing and jump straight to marshalling the json and printing it out

Reply:
Good spot, forgot about that.. changed under [35d3b3d](https://github.com/companieshouse/kafka-management-tool/pull/1/commits/35d3b3dbcc4359651dbe3fd05564be12fd6b6339)